res/controllers/Numark-Mixtrack-Platinum-FX-scripts.js

```diff
@@ -0,0 +1,1899 @@
+
+// state variable, don't touch
+MixtrackPlatinumFX.shifted = false;
+
+MixtrackPlatinumFX.initComplete=false;
+
+MixtrackPlatinumFX.bpms = []; 
+MixtrackPlatinumFX.trackBPM = function(value, group, _control) {
+	// file_bpm always seems to be 0?
+	// this doesn't work if we have to scan for bpm as it will be zero initially
+	// so we hook into the bpm change as well, and if we have 0 then set it to the first value seen (in bpm output)
+	MixtrackPlatinumFX.bpms[script.deckFromGroup(group) - 1] = engine.getValue(group,"bpm");
+}
+
+MixtrackPlatinumFX.BlinkTimer=0;
+MixtrackPlatinumFX.BlinkState=true;
+MixtrackPlatinumFX.BlinkStateSlow=true;
+MixtrackPlatinumFX.CallBacks=[];
+MixtrackPlatinumFX.CallSpeed=[];
+MixtrackPlatinumFX.BlinkStart = function(callback, slow) {
+	for (var i in MixtrackPlatinumFX.CallBacks) {
+		if (!MixtrackPlatinumFX.CallBacks[i]) {
+			// empty slot
+			MixtrackPlatinumFX.CallBacks[i]=callback;
+			MixtrackPlatinumFX.CallSpeed[i]=slow;
+			return i+1;
+		}
+	}
+	var idx = MixtrackPlatinumFX.CallBacks.push(callback);
+	MixtrackPlatinumFX.CallSpeed[idx-1]=slow;
+	return idx;
+};
+MixtrackPlatinumFX.BlinkStop = function(index) {
+	MixtrackPlatinumFX.CallBacks[index-1]=null;
+};
+MixtrackPlatinumFX.BlinkFunc = function() {
+	// toggle the global blink variables
+	MixtrackPlatinumFX.BlinkState = !MixtrackPlatinumFX.BlinkState;
+	if (MixtrackPlatinumFX.BlinkState) {
+		MixtrackPlatinumFX.BlinkStateSlow = !MixtrackPlatinumFX.BlinkStateSlow;
+	}
+	
+	// if we should be blinking the fx, then call its function
+	if (MixtrackPlatinumFX.FxBlinkState) {
+		MixtrackPlatinumFX.FxBlinkUpdateLEDs();
+	}
+	// fire any callbacks
+	for (var i in MixtrackPlatinumFX.CallBacks) {
+		if (MixtrackPlatinumFX.CallBacks[i]) {
+			if (MixtrackPlatinumFX.CallSpeed[i]) {
+				if (MixtrackPlatinumFX.BlinkState) {
+					MixtrackPlatinumFX.CallBacks[i](MixtrackPlatinumFX.BlinkStateSlow);
+				}
+			} else {
+				MixtrackPlatinumFX.CallBacks[i](MixtrackPlatinumFX.BlinkState);
+			}
+		}
+	}
+};
+
+MixtrackPlatinumFX.init = function(id, debug) {     
+    MixtrackPlatinumFX.id = id;
+    MixtrackPlatinumFX.debug = debug;
+    // print("init MixtrackPlatinumFX " + id + " debug: " + debug);
+    
+    // disable demo lightshow
+    var exitDemoSysex = [0xF0, 0x7E, 0x00, 0x06, 0x01, 0xF7];
+    midi.sendSysexMsg(exitDemoSysex, exitDemoSysex.length);
+
+// status, extra 04 is just more device id, not sure what the 05 is
+//F0 00 20 04 7F 03 01 05 F7
+
+// wake (not sure what the extra 07 is for?
+//F0 7E 00 07 06 01 F7
+
+// I think these are the dial updates
+//F0 00 20 04 7F 02 02 04 08 00 00 04 00 00 00 05 F7
+//F0 00 20 04 7F 04 01 04 00 00 00 04 00 00 00 05 F7
+//F0 00 20 04 7F 02 04 04 08 00 00 04 00 00 00 07 00 00 F7
+//F0 00 20 04 7F 03 02 04 08 00 00 04 00 00 00 05 F7
+//F0 00 20 04 7F 03 04 04 08 00 00 04 00 00 00 07 00 00 F7
+//F0 00 20 04 7F 01 04 04 08 00 00 04 00 00 00 07 00 00 F7
+//F0 00 20 04 7F 04 04 04 08 00 00 04 00 00 00 07 00 00 F7
+
```

Comment:
not sure how relevant this is for you, but I've reverse-engineered this all already. https://github.com/mixxxdj/mixxx/pull/11075/files#diff-39ced933348af25b914ee7fcd0d44355bb001d6151fb8781d7aca1471e304d52R3-R42

Reply:
I will need more research or help on this.

Reply:
The comment now just posts some messages, the link I shared has some extra info that explains them in better detail. I think some of this code could be better rewritten by salvaging code from that mapping, such as `NS6II.numberToSysex` and `NS6II.Display`.